apis/kueue/v1beta1/multikueueconfig_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MultiKueueConfigSpec defines the desired state of MultiKueueConfig
+type MultiKueueConfigSpec struct {
+	// clusters contains the list of configurations for using worker clusters.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	Clusters []MultiKueueCluster `json:"clusters"`
+}
+
+type MultiKueueCluster struct {
```

Comment:
The cluster configuration should be a top-level object as in the KEP for reusability, it is acceptable as in the KEP, but we should follow up with aligning the API with KEP in this regard. cc @mwielgus